app/models/assignment.rb

```diff
@@ -1199,6 +1199,35 @@ def assignment_properties_config
     end
   end
 
+  # Writes this assignment's starter files and settings to the +dir_name+ located in the +zip_file+.
+  # The settings for this assignment's starter files will be stored in a file located at +settings_filepath+
+  def starter_file_config_to_zip(zip_file, dir_name, settings_filepath)
+    default_starter_group = nil
+    download_count = 1
+    group_data = []
+    self.starter_file_groups.find_each do |starter_file_group|
+      group_name = "#{download_count}: #{starter_file_group.name}"
+      starter_file_group.write_starter_files_to_zip(zip_file, File.join(dir_name, group_name))
+      if starter_file_group.id == self.default_starter_file_group_id
+        default_starter_group = group_name
+      end
+      group_data << {
+        directory_name: group_name,
+        name: starter_file_group.name,
+        use_rename: starter_file_group.use_rename,
+        entry_rename: starter_file_group.entry_rename
+      }
+      download_count += 1
+    end
+    starter_file_settings = {
+      starter_file_type: self.starter_file_type,
+      allow_starter_files_after_due: self.starter_files_after_due,
```

Comment:
Shouldn't these two settings already be set in the assignment properties file?

Reply:
I have decided to move those here just so that related settings are group together